### ISSUE TITLE
Linux version (event pipe based)

### ISSUE THREAD

**Issue by alex-jitbit** [xoofx/ultra#15]
The tool looks amazing, please consider building Enent-Pipe based version for linux/macos (instead of ETW)

**Comment by xoofx** [comment]
> please consider building Enent-Pipe based version for linux/macos (instead of ETW)

It is actually not that simple, otherwise I would have done it in the first place 🙂

One of the main features of ultra is to provide native/kernel call stacks mixed with managed call-stacks, but Event-Pipe does not provide native call stacks on non-Windows platform. On Windows, it is provided by ETW, so it was easy to bring support for that (that's why the current codebase of ultra is so small for a "profiler" 😅)

I'm evaluating how to bring support for native call stacks on MacOS ARM64 first.

Then I will try Linux.

**Comment by alex-jitbit** [comment]
Thanks for looking into this. Even having managed-only traces would be a huge win, similar to what `dotnet-trace` does. Most people's use case is profiling in production, and most production runs on linux these days.

**Comment by xoofx** [comment]
> Most people's use case is profiling in production, and most production runs on linux these days.

Yeah, that really depends on the industry you are working in. 🙂 
In the gaming industry for example, most of the production are on mobile/desktop/console, and most of the development time is spent on Windows (90%) / macOS (~10%) / Linux, with Linux representing probably less than 1%.

**Comment by alex-jitbit** [comment]
@xoofx  Yeah. Definitely. Sorry, keep forgetting how big the Unity-verse is. In my defense: your awesome tool was recommended at Nick CHapsas's youtube, where he mostly profiles webservers.
